assets/charts/control-plane/kube-apiserver/templates/kube-apiserver.yaml

```diff
@@ -1,57 +1,23 @@
-# On platforms like AWS, it's enough for kube-apiserver to listen on Node InternalIP. On other
-# like Packet, we need to listen on all interfaces to be able to expose kube-apiserver to the
-# internet. As we cannot listen on specific IP address and on 0.0.0.0 on the same port when using
-# SO_REUSEPORT, kube-apiserver process must listen on different port than HAProxy does. So if we
-# need to expose kube-apiserver on all interfaces, kube-apiserver will be listening on random
-# 127.0.0.0/8 IP address on port 7443, but will advertise Node InternalIP with port 7443 and HAProxy
-# will be listening on both Node InternalIP on port 7443 and on 0.0.0.0:6443 for both internal and
-# external traffic.
-# See commit 95af6c96 for details.
-{{ define "port" }}
-{{- if and .Values.apiserver.exposeOnAllInterfaces (eq (int .Values.apiserver.replicas) 1) -}}
-7443
-{{- else -}}
-6443
-{{- end -}}
-{{ end }}
 {{- define "authHeader" -}}
 Bearer {{ template "token" . }}
 {{- end }}
 apiVersion: apps/v1
-# If there is just one controller node, we want to use Deployment to be able to run 2 kube-apiserver
-# pods on a single node at a time, to provide graceful upgrades.
-# See commit 95af6c96 for details.
-{{- if eq (int .Values.apiserver.replicas) 1 }}
-kind: Deployment
-{{- else }}
 kind: DaemonSet
-{{- end }}
 metadata:
   name: kube-apiserver
   namespace: kube-system
   labels:
     tier: control-plane
     k8s-app: kube-apiserver
 spec:
-  {{- if eq (int .Values.apiserver.replicas) 1 }}
-  replicas: 1
-  {{- end }}
   selector:
     matchLabels:
       tier: control-plane
       k8s-app: kube-apiserver
-  {{- if eq (int .Values.apiserver.replicas) 1 }}
-  strategy:
-  {{- else }}
   updateStrategy:
-  {{- end }}
     type: RollingUpdate
     rollingUpdate:
-      {{- if eq (int .Values.apiserver.replicas) 1 }}
-      maxUnavailable: 0
```

Comment:
This will cause an outage during cluster upgrade with single controller.

Reply:
added this as a default setting.